xdsl/irdl/declarative_assembly_format.py

```diff
@@ -120,7 +116,7 @@ def parse(
             stmt.parse(parser, state)
 
         # Get constraint variables from the parsed operand and result types
-        self.resolve_constraint_variables(state)
+        self.resolve_constraint_variables(state, op_def)
```

Comment:
This seems like a good place to catch any errors during extraction and re-raise it as a parsing error, with the start/end pos after parsing `self.stmts`?